### PR TITLE
Issue #4443 - Track backport of ALPN APIs to Java 8.

### LINKED ISSUE #4443: Track backport of ALPN APIs to Java 8
**Jetty version**
9.4.x

**Java version**
8

**Description**
Oracle is planning to backport to Java 8 the ALPN APIs that were introduced in Java 9, see https://mail.openjdk.java.net/pipermail/jdk8u-dev/2019-November/010573.html.

This will probably land on 8u252, so from that release onward we will need to change the ALPN providers for Java 9+ to work also on 8u252+.

This may require some code changes, as well as Jetty module changes.

### PR DESCRIPTION
**Pull request by sbordet**
Modified jetty-alpn-openjdk8-* classes to support both
pre 8u252 (via alpn-boot) and post 8u252 (via standard API).

Replaced usages of -Xbootclasspath with -javaagent, and
using Jetty ALPN Agent jar rather than Jetty ALPN boot jar.

Removed all alpn-1.8.0*.mod files since now it is
possible to use a fixed version of the ALPN Agent
to cover all the versions.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

Closes #4443.

### CODE REVIEW

**Comment by gregw** [comment]
@sbordet this PR has conflicts and errors.

**Comment by sbordet** [comment]
@gregw I pushed the fix for the conflicts, the failure is unrelated (some problem stopping mongodb in the session tests).

**Comment by olamy** [comment]
build has failed because of 2h timeout. Not sure why the build took so long. next builds were ok

File: jetty-alpn/jetty-alpn-openjdk8-client/src/main/java/org/eclipse/jetty/alpn/openjdk8/client/OpenJDK8ClientALPNProcessor.java [hunk 1]
```diff
@@ -33,11 +40,28 @@
...
+        try
+        {
+            // JDK 8u252 has the JDK 9 ALPN API backported.
+            // Use reflection so we can build with a JDK version less than 8u252.
+            alpnProtocols = SSLParameters.class.getMethod("setApplicationProtocols", String[].class);
+            alpnProtocol = SSLEngine.class.getMethod("getApplicationProtocol");
+            return;
+        }
+        catch (NoSuchMethodException ignored)
+        {
```
Comment (gregw) [inline jetty-alpn/jetty-alpn-openjdk8-client/src/main/java/org/eclipse/jetty/alpn/openjdk8/client/OpenJDK8ClientALPNProcessor.java:63]: Probably should do a LOG.ignore here so we can see the exception if we really want to

File: jetty-alpn/jetty-alpn-openjdk8-client/src/main/java/org/eclipse/jetty/alpn/openjdk8/client/OpenJDK8ClientALPNProcessor.java [hunk 2]
```diff
@@ -33,11 +40,28 @@
...
         if (JavaVersion.VERSION.getPlatform() != 8)
             throw new IllegalStateException(this + " not applicable for java " + JavaVersion.VERSION);
+
+        try
+        {
+            // JDK 8u252 has the JDK 9 ALPN API backported.
+            // Use reflection so we can build with a JDK version less than 8u252.
+            alpnProtocols = SSLParameters.class.getMethod("setApplicationProtocols", String[].class);
+            alpnProtocol = SSLEngine.class.getMethod("getApplicationProtocol");
+            return;
```
Comment (gregw) [inline jetty-alpn/jetty-alpn-openjdk8-client/src/main/java/org/eclipse/jetty/alpn/openjdk8/client/OpenJDK8ClientALPNProcessor.java:60]: Probably should do a LOG.debug here

**Review by gregw — APPROVED** [review]
LGTM but a few quibbles about better logging

File: jetty-alpn/jetty-alpn-openjdk8-client/src/main/java/org/eclipse/jetty/alpn/openjdk8/client/OpenJDK8ClientALPNProcessor.java (hunk 1 above)
Comment (sbordet) [reply jetty-alpn/jetty-alpn-openjdk8-client/src/main/java/org/eclipse/jetty/alpn/openjdk8/client/OpenJDK8ClientALPNProcessor.java:63]: Done

File: jetty-alpn/jetty-alpn-openjdk8-client/src/main/java/org/eclipse/jetty/alpn/openjdk8/client/OpenJDK8ClientALPNProcessor.java (hunk 2 above)
Comment (sbordet) [reply jetty-alpn/jetty-alpn-openjdk8-client/src/main/java/org/eclipse/jetty/alpn/openjdk8/client/OpenJDK8ClientALPNProcessor.java:60]: Done

**Pull request merged**